mypyc/primitives/bytes_ops.py

```diff
@@ -19,6 +19,15 @@
     c_function_name='PyBytes_FromObject',
     error_kind=ERR_MAGIC)
 
+# bytes(int)
+function_op(
+    name='builtins.bytes',
+    arg_types=[int_rprimitive],
+    return_type=bytes_rprimitive,
+    c_function_name='CPyBytes_FromInt',
+    error_kind=ERR_MAGIC,
+    priority=2)
```

Comment:
Using a priority to pick an alternative implementation isn't enough here, since the integer argument could have a static type `Any`, and this alternative implementation wouldn't be picked up. The integer case would need to be handled in the main primitive that accepts any `object`. Also, since `bytes(n)` is an extremely rare thing to do, maybe you shouldn't bother with it and only optimize the more common cases.

Reply:
I prefer using RUnion now